File: java/Observer/src/observer/BreakingNewsFeed.java
package observer;

import java.util.ArrayList;
import java.util.List;

public class BreakingNewsFeed {
	String breakingNews;
	
	List<Observer> observers = new ArrayList<>();
	
	public void subscribe(Observer observer){
		observers.add(observer);
	}

	public String getBreakingNews() {
		return breakingNews;
	}

	public void setBreakingNews(String breakingNews) {
		this.breakingNews = breakingNews;
		notifySubsrcribers(); // > 1
	}
	
	private void notifySubsrcribers() {
		for(Observer observer : observers)
			observer.sendNotification(); // > 2
	}
}

File: ruby/Observer/breaking_news_feed.rb
class BreakingNewsFeed
  attr_accessor :breaking_news

  def initialize
    @observers = Array.new
  end

  def subscribe (observer)
    @observers.push(observer)
  end

  def notify_subscriber
    for observer in @observers
      observer.sendNotification # > 1
    end
  end

  def set_breaking_news (breaking_news)
    @breaking_news = breaking_news
    notify_subscriber # > 2
  end
end